### PR TITLE
feat: add a github client

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Add a github client to retrieve pull request status from a repository.
- Add unit test.

### CODE REVIEW

File: java-shared-dependencies/dependency-analyzer/pom.xml [hunk 1]
```diff
@@ -38,6 +38,16 @@
           </arguments>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.2.5</version>
+        <configuration>
+          <environmentVariables>
+            <GITHUB_TOKEN>fake_value</GITHUB_TOKEN>
```
Comment (suztomo) [inline java-shared-dependencies/dependency-analyzer/pom.xml:47]: Add source code comment how it helps tests.

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class GitHubClient {
```
Comment (suztomo) [inline java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java:23]: Can you add Javadoc?

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/DepsDevClient.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 public class DepsDevClient {
```
Comment (suztomo) [inline java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/DepsDevClient.java:29]: Can you add Javadoc?

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/PullRequestStatus.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.google.cloud.model;
+
+public record PullRequestStatus(long created, long merged, Interval interval) {
```
Comment (suztomo) [inline java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/PullRequestStatus.java:3]: Can you add Javadoc, especially which part of the dependency report this helps, with an example value.

**Review by suztomo — COMMENTED** [review]
Would you add brief Javadoc for future maintainers?

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/PullRequestStatus.java (hunk 4 above)
Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/PullRequestStatus.java:3]: We can change the class name to `PullRequestStatistics`.

Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/PullRequestStatus.java]: Done.

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/DepsDevClient.java (hunk 3 above)
Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/DepsDevClient.java:37]: Done.

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java (hunk 2 above)
Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java:33]: Done.

File: java-shared-dependencies/dependency-analyzer/pom.xml (hunk 1 above)
Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/pom.xml:49]: Done.

**Review by suztomo — APPROVED** [review]
Can you run the [google-java-format formatter](https://github.com/google/google-java-format)?

`git diff -U0 origin/main |google-java-format-diff.py -p1 -i`

The formatter does not merge multiple lines. Ensure you read the formatted files.

File: java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * GitHubClient is a class that sends HTTP requests to the GitHub RESTful API.
+ * It provides methods for interacting with various GitHub resources such as repositories,
+ * issues, users, etc.
+ * <p>
```
Comment (suztomo) [inline java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java:27]: New line before `<p>`. No new line after `<p>`. https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs

Comment (JoeWang1127) [reply java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/external/GitHubClient.java]: I formatted the code using
```
git diff -U0 origin/main | python google-java-format-diff.py -p1 -i
```

**Pull request merged**